service/controller/clusterapi/v22/resource/clusterconfigmap/desired.go

```diff
@@ -19,12 +20,26 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) ([]*cor
 		return nil, microerror.Mask(err)
 	}
 
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	var workerCount int32
+
+	for _, nodePool := range cc.Status.Worker {
+		workerCount += nodePool.Nodes
```

Comment:
The controller context exposes both the node count and the ready nodes.

My worry is if we don't use ready then for a new node pool multiple nodes will be added at once and the deployment will be unable to schedule the pods until the nodes are ready.

Should we add the nodes for the first node pool but then only ready nodes for the other node pools?


https://github.com/giantswarm/cluster-operator/blob/14bd41618c35eb3fe24b34e269a8d167b752bc61/service/controller/clusterapi/v22/controllercontext/status.go#L17-L20